### PR TITLE
mode/prompt-buffer..watch: Review documentation.

### PR DESCRIPTION
**Pull request by Ambrevar**
# Description

Follow up and final of the mode documentation review started with #2957 and #2959.

Note that unlike @aartaka I haven't extensively listed the commands in the mode docstring.  We can always address this later when we reach a consensus.

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.

- [x] Git hygiene:
  - I have pulled from master before submitting this PR
  - There are no merge conflicts.
- [x] I've added the new dependencies as:
  - ASDF dependencies,
  - Git submodules,
    ```sh
	cd /path/to/nyxt/checkout
    git submodule add https://gitlab.common-lisp.net/nyxt/py-configparser _build/py-configparser
    ```
  - and Guix dependencies.
- [x] My code follows the style guidelines for Common Lisp code. See:
  - [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf)
  - [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml)
- [ ] I have performed a self-review of my own code.
- [ ] My code has been reviewed by at least one peer.  (The peer review to approve a PR counts.  The reviewer must download and test the code.)
- [x] Documentation:
  - All my code has docstrings and `:documentation`s written in the aforementioned style.  (It's OK to skip the docstring for really trivial parts.)
  - I have updated the existing documentation to match my changes.
  - I have commented my code in hard-to-understand areas.
  - I have updated the `changelog.lisp` with my changes if it's anything user-facing (new features, important bug fix, compatibility breakage).
  - I have added a `migration.lisp` entry for all compatibility-breaking changes.
  - (If this changes something about the features showcased on Nyxt website) I have these changes described in the new/existing article at Nyxt website or will notify one of maintainters to do so.
- [x] Compilation and tests:
  - My changes generate no new warnings.
  - I have added tests that prove my fix is effective or that my feature works.  (If possible.)
  - I ran the tests locally (`(asdf:test-system :nyxt)` and `(asdf:test-system :nyxt/gi-gtk)`) and they pass.

### CODE REVIEW

File: source/mode/style.lisp [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ;;;; SPDX-License-Identifier: BSD-3-Clause
 
 (nyxt:define-package :nyxt/mode/style
-    (:documentation "Mode for styling documents."))
+  (:documentation "Package for `style-mode', a customizable document styling facility."))
```
Comment (aartaka) [inline source/mode/style.lisp:5]: I'd also say that it hosts `dark-mode'.

File: source/mode/vi.lisp [hunk 2]
```diff
@@ -2,7 +2,7 @@
 ;;;; SPDX-License-Identifier: BSD-3-Clause
 
 (nyxt:define-package :nyxt/mode/vi
-  (:documentation "VI-style bindings."))
+  (:documentation "Package for `vi-normale-mode' and `vi-insert-mode', which provide VI-style bindings."))
```
Comment (aartaka) [inline source/mode/vi.lisp:5]: You mean "vi-normal~e~-mode?

**Review by aartaka — APPROVED** [review]
I mean yeah, we can add things later, but would be nice to have them ASAP. Anywas, feel free to merge, it's all good except for some bits I've commented on.

**Comment by Ambrevar** [comment]
> I mean yeah, we can add things later, but would be nice to have them ASAP.

I'm confused, you just said yourself that you agreed we should not list the commands.  What do you mean with "them"?

File: source/mode/style.lisp (hunk 1 above)
Comment (Ambrevar) [reply source/mode/style.lisp]: Done.

**Comment by aartaka** [comment]
> > I mean yeah, we can add things later, but would be nice to have them ASAP.
> 
> I'm confused, you just said yourself that you agreed we should not list the commands. What do you mean with "them"?

I mean listing APIs in general, not commands. More like internal functions/helpers.

**Comment by aartaka** [comment]
> > More like internal functions/helpers.
> 
> You mean "exported"? I don't believe the package docstring should list unexported helpers.

Mostly yes, but I also list the internal ones too, because some of them are useful.

**Comment by Ambrevar** [comment]
> Mostly yes, but I also list the internal ones too, because some of them are useful.

Hmmm...  We should be careful with this.  At the very least, they should have a `PACKAGE::` prefix.

The contact is that only exported symbols are a stable API.  Unexported symbols are prone to change.  And really, if they are useful to the user, they should be exported.

**Comment by aartaka** [comment]
Fair, yes.

**Comment by aartaka** [comment]
Anyway, it's OK to merge this!

**Comment by aartaka** [comment]
Thanks!

**Pull request merged**